app/scripts/museum/museum-service.js

```diff
@@ -100,6 +106,30 @@
             return Util.makeRequest(sql, query);
         }
 
+        function listByCity(params) {
+            var attributes = { 'gcity': 'city', 'gstate': 'state', 'gzip': 'zip' };
```

Comment:
Oh I see. An expression won't be generated for `zip` unless it is included in `params`.